engine/src/test/java/io/camunda/zeebe/engine/processing/resource/ResourceDeletionRejectionTest.java

```diff
@@ -35,4 +39,62 @@ public void shouldRejectCommandWhenResourceIsNotFound() {
         .hasRejectionReason(
             "Expected to delete resource but no resource found with key `%d`".formatted(key));
   }
+
+  @Test
+  public void shouldRejectCreateInstanceCommandWhenOnlyProcessVersionIsDeleted() {
+    // given
+    final var processId = helper.getBpmnProcessId();
+    final var processDefinitionKey = deployProcess(processId);
+    engine.resourceDeletion().withResourceKey(processDefinitionKey).delete();
+
+    // when
+    engine.processInstance().ofBpmnProcessId(processId).expectRejection().create();
+
+    // then
+    final var rejectionRecord =
+        RecordingExporter.processInstanceCreationRecords().onlyCommandRejections().getFirst();
+
+    Assertions.assertThat(rejectionRecord)
+        .hasIntent(ProcessInstanceCreationIntent.CREATE)
+        .hasRejectionType(RejectionType.NOT_FOUND)
+        .hasRejectionReason(
+            String.format(
+                "Expected to find process definition with process ID '%s', but none found",
+                processId));
+  }
```

Comment:
👍 Good test case to have!

🔧 I'd argue this should be in the `ResourceDeletionTest` as this is part of successful deletion

Reply:
I'd argue otherwise as we are verifying the rejection. The `ResourceDeletionTest` contains cases that verify a successful deletion already.

Reply:
🤔 But isn't this exactly a successful deletion? As part of a successful deletion, we want to make it impossible to create instances of the deleted process.